code/modules/bitrunning/objects/netpod.dm

```diff
@@ -315,7 +315,13 @@
 		if(isnull(wayout))
 			balloon_alert(neo, "out of bandwidth!")
 			return
-		current_avatar = server.generate_avatar(wayout, netsuit)
+		// BUBBER EDIT BEGIN - PREFS!
+		var/datum/preferences/pref
+		for(var/obj/item/bitrunning_disk/prefs/prefdisk in neo.get_contents())
+			pref = prefdisk.loaded_preference
+			current_avatar = server.generate_avatar(wayout, netsuit, pref, include_loadout = prefdisk.include_loadout)  // Added the prefs argument
+			break
```

Comment:
So, uh, what if there is no prefdisk? Is that a valid state?

Reply:
Also the way this is written it may call generate_avatar multiple times, is that a good idea?

Reply:
> So, uh, what if there is no prefdisk? Is that a valid state?

Shit

> Also the way this is written it may call generate_avatar multiple times, is that a good idea?

see: break

Reply:
Anyhow, fixed